lib/ecl/ecl_sum_file_data.cpp

```diff
@@ -0,0 +1,736 @@
+  return this->time_range.second;
+}
+
+time_t ecl_sum_file_data::iget_sim_time(int time_index) const {
+  const ecl_sum_tstep_type * ministep = iget_ministep( time_index  );
+  return ecl_sum_tstep_get_sim_time(ministep);
+}
+
+double ecl_sum_file_data::iget_seconds(int time_index) const {
+  throw;
+  return 0;
+}
+
+double ecl_sum_file_data::get_sim_length() const {
+  return this->sim_length;
+}
+
+double ecl_sum_file_data::iget( int time_index , int params_index ) const {
+  if (params_index >= 0) {
+    const ecl_sum_tstep_type * ministep_data = iget_ministep( time_index  );
+    return ecl_sum_tstep_iget( ministep_data , params_index); 
+  }
+  else
+    return 0;
+}
+
+
+
+void ecl_sum_file_data::clear_index() {
+  this->report_map.clear();
+  first_report_step     =  1024 * 1024;
+  last_report_step      = -1024 * 1024;
+  days_start            = 0;
+  sim_length            = -1;
+  index_valid           = false;
+  time_range            = std::make_pair<time_t, time_t>(INVALID_TIME_T, INVALID_TIME_T);
+}
+
+
+void ecl_sum_file_data::append_tstep(ecl_sum_tstep_type * tstep) {
+  /*
+     Here the tstep is just appended naively, the vector will be
+     sorted by ministep_nr before the data instance is returned.
+  */
+
+  vector_append_owned_ref( data , tstep , ecl_sum_tstep_free__);
+  index_valid = false;
+}
+
+
+/*
+  This function is meant to be called in write mode; and will create a
+  new and empty tstep which is appended to the current data. The tstep
+  will also be returned, so the calling scope can call
+  ecl_sum_tstep_iset() to set elements in the tstep.
+*/
+
+ecl_sum_tstep_type * ecl_sum_file_data::add_new_tstep( int report_step , double sim_seconds) {
+  int ministep_nr = vector_get_size( data );
+  ecl_sum_tstep_type * tstep = ecl_sum_tstep_alloc_new( report_step , ministep_nr , sim_seconds , ecl_smspec );
+  ecl_sum_tstep_type * prev_tstep = NULL;
+
+  if (vector_get_size( data ) > 0)
+    prev_tstep = (ecl_sum_tstep_type*)vector_get_last( data );
+
+  append_tstep( tstep );
+  {
+    bool rebuild_index = true;
+
+    /*
+      In the simple case that we just add another timestep to the
+      currently active report_step, we do a limited update of the
+      index, otherwise we call ecl_sum_data_build_index() to get a
+      full recalculation of the index.
+    */
+
+    if (prev_tstep != NULL) {
+      if (ecl_sum_tstep_get_report( prev_tstep ) == ecl_sum_tstep_get_report( tstep )) {        // Same report step
+        if (ecl_sum_tstep_get_sim_days( prev_tstep ) < ecl_sum_tstep_get_sim_days( tstep )) {   // This tstep will become the new latest tstep
+          int internal_index = vector_get_size( data ) - 1;
+          this->report_map[report_step].second = internal_index;
+
+          sim_length = ecl_sum_tstep_get_sim_days( tstep );
+          this->time_range.second = ecl_sum_tstep_get_sim_time(tstep);
+
+          rebuild_index = false;
+        }
+      }
+    }
```

Comment:
I won't take your guard away from you if you really want it there, but I really think that the following is easier (fewer nested `if`s).

```c++
bool need_rebuild =
     prev_tstep == NULL
  || ecl_sum_tstep_get_report(prev_tstep) != ecl_sum_tstep_get_report(tstep))
  || ecl_sum_tstep_get_sim_days(prev_tstep) >= ecl_sum_tstep_get_sim_days(tstep);


if (need_rebuild) {
  int internal_index = vector_get_size( data ) - 1;
  this->report_map[report_step].second = internal_index;

  sim_length = ecl_sum_tstep_get_sim_days( tstep );
  this->time_range.second = ecl_sum_tstep_get_sim_time(tstep);

  rebuild_index = false;
}
```

Reply:
I agree - that code was moved unchanged.